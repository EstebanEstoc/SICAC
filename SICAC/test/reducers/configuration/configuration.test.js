import configurationReducer from '../../../reducers/configuration/configrationSlice'
import {
  modifyHouseWifi,
  setRelativesNumbers,
  modifyDoctoreNumber,
  modifyDefaultCalendarId,
  modifySmarWatch,
  modifyCardiacThreshold,
  clearHouseWifi,
  clearRelativesNumbers,
  clearDoctorNumber,
  clearSmartWatch,
  clearCardiacThreshold,
  clearDefaultCalendarID,
  clearConfig
} from '../../../reducers/configuration/configrationSlice'

describe('configuration Reducer', () => {
  describe('INITIAL_STATE', () => {
    test('is correct', () => {
      const action = { type: 'dummy_action' }
      const initialState = {
        houseWifi: '',
        relativesNumbers: [],
        doctorNumber: '',
        smartWatch: '',
        cardiacThreshold: 180,
        defaultCalendarID: ''
      }
      expect(configurationReducer(undefined, action)).toEqual(initialState)
    })
  })

  describe('modifyHouseWifi', () => {
    test('must add the Wifi SSID to houseWifi', () => {
      const modifyHouseWifiPayload = 'wifi_SSID_test'
      const action = {
        type: modifyHouseWifi.type,
        payload: modifyHouseWifiPayload
      }
      expect(configurationReducer(undefined, action)).toMatchSnapshot()
    })
  })

  describe('setRelativesNumbers', () => {
    test('must add Relatives Numbers to relativesNumbers', () => {
      const setRelativesNumbersPayload = ['+33606060606', '+33606060606']
      const action = {
        type: setRelativesNumbers.type,
        payload: setRelativesNumbersPayload
      }
      expect(configurationReducer(undefined, action)).toMatchSnapshot()
    })
  })
})
