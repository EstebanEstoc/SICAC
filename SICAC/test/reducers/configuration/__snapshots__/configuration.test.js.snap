// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configuration Reducer clearHouseWifi must add the Wifi SSID to houseWifi 1`] = `
Object {
  "cardiacThreshold": 180,
  "defaultCalendarID": "",
  "doctorNumber": "",
  "houseWifi": "",
  "relativesNumbers": Array [],
  "smartWatch": "",
}
`;

exports[`configuration Reducer modifyHouseWifi must add the Wifi SSID to houseWifi 1`] = `
Object {
  "cardiacThreshold": 180,
  "defaultCalendarID": "",
  "doctorNumber": "",
  "houseWifi": "wifi_SSID_test",
  "relativesNumbers": Array [],
  "smartWatch": "",
}
`;

exports[`configuration Reducer setRelativesNumbers must add Relatives Numbers to relativesNumbers 1`] = `
Object {
  "cardiacThreshold": 180,
  "defaultCalendarID": "",
  "doctorNumber": "",
  "houseWifi": "",
  "relativesNumbers": Array [
    "+33606060606",
    "+33606060606",
  ],
  "smartWatch": "",
}
`;
